src/H5FDmpio.c

```diff
@@ -3099,7 +3099,7 @@ H5FD__mpio_read_selection(H5FD_t *_file, H5FD_mem_t type, hid_t H5_ATTR_UNUSED d
              */
             j = 0; /* guess at the index of the smallest value of s_bufs[] */
 
-            if (s_bufs[j + 1].vp != NULL) {
+            if ((count > 1) && (s_bufs[j + 1].vp != NULL)) {
```

Comment:
Is the check for s_bufs[j + 1].vp (which should always reduce to s_bufs[1].vp) actually necessary here? If not, can we just eliminate this if statement entirely since the for loop exit condition should cover it?

Reply:
@vchoi-hdfgroup

Reply:
After looking at this, yes I believe the if is unnecessary due to the for loop conditions.

Reply:
Actually, after more investigation it looks like the check is needed because there is some form of optimization being applied where entries in `s_bufs` are being left uninitialized and so the `if (s_bufs[j + 1].vp != NULL)` check is used to check for this optimization before checking any further into the array.